### ISSUE TITLE
RCTFatalException: Unhandled JS Exception: TypeError: Super expression must either be null or a function, js engine: hermes

### ISSUE THREAD

**Issue by efstathiosntonas** [segmentio/analytics-react-native#892]
- `analytics-react-native` version: 2.17.0
- Integrations versions (if used):
- React Native version: 0.72.6
- iOS or Android or both? 

**Steps to reproduce**

use version 2.17.0 and build a release variant, app insta crashes with: 

```
RCTFatalException: Unhandled JS Exception: TypeError: Super expression must either be null or a function, js engine: hermes
```

downgrading to 2.16.1 solves the issue. When on debug variant it works fine, it only happens in release variant. Let me know if you need any additional info. 

```
    "@babel/core": "^7.23.2",
    "@babel/preset-env": "7.23.2",
    "@babel/preset-typescript": "^7.23.2",
    "@babel/runtime": "^7.23.2",
    "typescript": "^5.2.2"
```

**Comment by BenhurKulzer** [comment]
Same error here, tryed to downgrade to older versions but still getting the error:

`TypeError: Super expression must either be null or a function, js engine: hermes`

**Comment by efstathiosntonas** [comment]
@BenhurKulzer make sure there's no ^ before the version on your package.json.

**Comment by efstathiosntonas** [comment]
Why issues are so slow in resolving them? This issue is breaking version 2.17 entirely and it's open for 8 days. Is a multi million company lacking resources on resolving issues fast? We're paying thousands of $ every single month and the support is terrible. Yes, it's open source but damn, you're making tons of money from it.

**Comment by danielamorimp** [comment]
I've been having the same problem for two weeks, it completely delays the releases. Did anyone manage to solve it?

**Comment by danielamorimp** [comment]
I've been having the same problem for two weeks, it completely delays the releases. Did anyone manage to solve it?

**Comment by oscb** [comment]
No good excuses why no one took a look at this during time off. Getting back here and looking at this.

**Comment by oscb** [comment]
Still checking. I have removed meanwhile the faulty version from NPM.

**Comment by oscb** [comment]
@BenhurKulzer @danielamorimp @efstathiosntonas Sorry for the lack of updates, we have been busy upgrading our tooling to 0.72+ to see if we could catch this but so far no luck. To be fair we're not deviating much from the defaults of a new RN app at each version we have tested (0.71.5, 0.72.6, 0.72.7). At this point we believe this might be caused by a common configuration we haven't tested. I'm thinking something along TS, Babel or RN configs/version combinations. 

Any chance someone can share a minimal repro project? Or your project dependencies and configurations for babel, rn, typescript?

**Comment by oscb** [comment]
Fix is released in v2.19
